### PR TITLE
augeas: workaround unfixed libtool bug

### LINKED ISSUE #83407: Augeas bottles for Big Sur are broken due to an outdated configure script
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 3.2.7-12-gbbd1954
ORIGIN: https://github.com/Homebrew/brew
HEAD: bbd1954523c986dc3e15096c571a2b913a290fac
Last commit: 21 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6944bf4e08d9f2fa49a3a938af9f39d04ad84f40
Core tap last commit: 8 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_NO_INSECURE_REDIRECT: set
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0.5 build 1205
Git: 2.32.0 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.5.1-arm64
CLT: 12.5.1.0.1.1623191612
Xcode: N/A
Rosetta 2: false
$ brew doctor
Your system is ready to brew.
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I work at EFF on Certbot which is packaged in Homebrew. Due to this problem in `augeas` which is one of Certbot's dependencies, [many of our users are reporting problems using Certbot on Big Sur](https://github.com/certbot/certbot/issues/8978#issue-963389257). This is my attempt to report the root problem and I apologize that these instructions are somewhat convoluted.

### What happened (include all command output)?

To reproduce the problem, you can run:
```
python3 -m venv venv
. ./venv/bin/activate
pip install python-augeas
python break.py
```
where `break.py` is:
```
from augeas import Augeas
a = Augeas(flags=Augeas.NONE | Augeas.NO_MODL_AUTOLOAD)
a.set('/augeas/load/Httpd/lens', 'Httpd.lns')
a.load()
print(a.get('/augeas/load/Httpd/error'))
```
The output of the last command is:
```
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/util.aug:115.2-123.51:Failed to compile comment_multiline
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/util.aug:116.20-.69:exception: Regular expression subtraction 'r1 - r2' failed
    r1: /[^ 	
].*[^ 	
]|[^ 	
]/
    r2: /.*\*/.*/

/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:78.26-.37:Could not load module Util for Util.indent
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:78.26-.37:Undefined variable Util.indent
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:168.4-.10:Undefined variable indent
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:177.13-.19:Undefined variable indent
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:182.21-.27:Undefined variable indent
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:187.18-.25:Undefined variable section
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:187.26-.44:Undefined variable directive
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:190.10-.18:Undefined variable content
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:204.13-.25:Undefined variable Util.stdexcl
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:206.20-.23:Undefined variable lns
/opt/homebrew/Cellar/augeas/1.12.0/share/augeas/lenses/dist/httpd.aug:206.24-.30:Undefined variable filter
```

### What did you expect to happen?

`python break.py` in the above section should just output `None`. This is the behavior you should see on versions of macOS other than Big Sur or if you install `augeas` on Big Sur using `brew install augeas --head`. I've reproduced this problem both on Intel and M1 Macs running Big Sur. The error you see in my output above is Augeas failing to parse its own [lenses](https://augeas.net/docs/lenses.html).

I believe the root cause here is that the configure script in the latest Augeas release doesn't properly handle the macOS version number change from 10.x to 11. I reported this problem upstream a month ago at https://github.com/hercules-team/augeas/issues/729 and received no response. At least until Augeas does another release, I'd love it if Homebrew would patch its copy of the configure script to properly set the compiler flags to avoid this problem and to fix the issue in the Augeas bottles for Big Sur.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install augeas
brew install python
# Then run the commands in the "What happened" section
```

### PR DESCRIPTION
**Pull request by bmw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Closes #83407.

### CODE REVIEW

File: Formula/augeas.rb [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+    url "http://download.augeas.net/augeas-1.12.0.tar.gz"
+    sha256 "321942c9cc32185e2e9cb72d0a70eea106635b50269075aca6714e3ec282cb87"
+    # This patch is needed until the bug in GNU's libtool at
+    # https://debbugs.gnu.org/cgi/bugreport.cgi?bug=44605 is resolved, a
+    # patched version makes its way to the Augeas' maintainers, and they
+    # release a new version. When this patch is removed, the autoconf and
+    # automake build dependencies should only be needed when building from
+    # head. You can find more info at
+    # https://github.com/Homebrew/homebrew-core/issues/83407.
+    patch :DATA
```
[the remark this answers is not in the document]
Comment (bmw) [reply Formula/augeas.rb:14]: Any thoughts on the preferred path forward here?

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/augeas.rb:14]: I think their custom script is mostly for convenience. If we can get away with just regenerating configure that would be preferable to me.

[the remark this answers is not in the document]
Comment (bmw) [reply Formula/augeas.rb:14]: Thanks for the feedback. The latest commit implements that approach and still fixes https://github.com/Homebrew/homebrew-core/issues/83407.

**Comment by bmw** [comment]
Can I get another review here when someone gets the time?

**Review by SMillerDev — APPROVED** [review]
Thanks @bmw  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**